Beta docs (#900)
microsoftgraph/entra-powershell: module/docs/entra-powershell-beta/Microsoft.Graph.Entra.Beta/Set-EntraBetaAttributeSet.md

# Set-EntraBetaAttributeSet

## Synopsis

Updates an existing attribute set.

## Syntax

```powershell
Set-EntraBetaAttributeSet 
 -Id <String>
 [-Description <String>] 
 [-MaxAttributesPerSet <Int32>]
 [<CommonParameters>]
```

## Description

> Updates a Microsoft Entra ID attribute set object identified by ID. Specify `Id` parameter to update an attribute set.

Reviewer: ```suggestion
Updates a Microsoft Entra ID attribute set object identified by ID. Specify `Id` parameter to update an attribute set.

In delegated scenarios with work or school accounts, the signed-in user must be assigned a supported Microsoft Entra role or a custom role with the necessary permissions. The Attribute Definition Administrator is the only privileged role supported for this operation.
```

Reviewer 2: changes done

## Examples

### Example 1: Update an attribute set

```powershell
Connect-Entra -Scopes 'CustomSecAttributeDefinition.ReadWrite.All'
$params = @{
    Id = 'Testing' 
    Description = 'Attributes for engineering team'
}
Set-EntraBetaAttributeSet @params
```

This example update an attribute set.

> - `Id` parameter specifies the name of the attribute set.

Reviewer: ```suggestion
- `Id` parameter specifies the name of the attribute set. You can `Get-EntraBetaAttributeSet` to get more details.
```

Reviewer 2: changes done

- `Description` parameter specifies the description for the attribute set.

### Example 2: Update an attribute set using MaxAttributesPerSet

```powershell
Connect-Entra -Scopes 'CustomSecAttributeDefinition.ReadWrite.All'
$params = @{
    Id = 'Testing' 
    MaxAttributesPerSet = 10
}
Set-EntraBetaAttributeSet @params
```

This example update an attribute set using MaxAttributesPerSet.

> - `Id` parameter specifies the name of the attribute set.

Reviewer: ```suggestion
- `Id` parameter specifies the name of the attribute set. You can `Get-EntraBetaAttributeSet` to get more details.
```

Reviewer 2: changes done

- `MaxAttributesPerSet` parameter specifies the maximum number of custom security attributes.

## Parameters

### -Description

> Description for the attribute set.

Reviewer: ```suggestion
Description of the attribute set, up to 128 characters long, including Unicode characters. This description can be changed later.
```

Reviewer 2: changes done

```

### -Id

> Name of the attribute set. Must be unique within a tenant.

Reviewer: ```suggestion
Name of the attribute set. Unique identifier for the attribute set within a tenant. This identifier can be up to 32 characters long and may include Unicode characters. It cannot contain spaces or special characters, and it cannot be changed later. The identifier is case insensitive.
```

Reviewer 2: changes done

```

### -MaxAttributesPerSet

> Maximum number of custom security attributes that can be defined in the attribute set.

Reviewer: ```suggestion
Maximum number of custom security attributes that can be defined in this attribute set. The default value is null. If not specified, the administrator can add up to 500 active attributes per tenant. This setting can be changed later.
```

Reviewer 2: changes done
